pkg/proxy/iptables/proxier.go

```diff
@@ -764,7 +806,7 @@ func getLocalIPs(endpointsMap proxyEndpointsMap) map[types.NamespacedName]sets.S
 				if localIPs[nsn] == nil {
 					localIPs[nsn] = sets.NewString()
 				}
-				ip := strings.Split(ep.endpoint, ":")[0] // just the IP part
+				ip := ep.endpoint[0:strings.Index(ep.endpoint, ":")] //just the IP part
```

Comment:
If we were to start doing whole-IP services, which has been asked for, I could see endpoints maybe becoming just an IP.    I'd rather this NOT assume : is present (lest I end up having to do `1.2.3.4:0` or something.  How about `max(len(ep.endpoint), strings.Index(...`